### PR TITLE
Fix file upload adjustments

### PR DESCRIPTION
**Pull request by vogdb**
Followup of #2392

### CODE REVIEW

**Comment by vogdb** [comment]
@christopheschwyzer please review

**Comment by vogdb** [comment]
@njam please review

**Comment by vogdb** [comment]
@zazabe please review

**Comment by vogdb** [comment]
anyone?

**Comment by christopheschwyzer** [comment]
Code looks ok to me. 
But I'm wondering - If cardinality=1 we probably should just remove the _multiple_ attribute from the form input field - this way we would prevent uploading multiple files altogether. 
cc @njam

**Comment by njam** [comment]
*reviewing*

File: library/CM/FormField/File.js [hunk 1]
```diff
@@ -66,13 +67,20 @@ var CM_FormField_File = CM_FormField_Abstract.extend({
...
         if (data.result.success) {
           data.$preview.html(data.result.success.preview + '<input type="hidden" name="' + field.getName() + '[]" value="' + data.result.success.id + '"/>');
-          if (inProgressCount === 0) {
-            field.trigger("uploadComplete", data.files);
-          }
         } else if (data.result.error) {
           data.$preview.remove();
           field.error(data.result.error.msg);
         }
+        if (inProgressCount === 0) {
```
Comment (njam) [inline library/CM/FormField/File.js]: Why only check it when the last upload finishes? It means that I can upload 10 files to a field with the cardinality of 2, and the 8 too many will only be removed when all are successfully uploaded.
How about just remove them as they come in?

**Comment by vogdb** [comment]
@njam please look

**Comment by vogdb** [comment]
@njam please review once more, sorry

**Comment by njam** [comment]
ok lgtm

**Pull request merged**